virtualizarr/writers/icechunk.py

```diff
@@ -112,58 +124,145 @@ def write_variable_to_icechunk(
+        existing_array = group[name]
+        append_axis = get_axis(dims, append_dim)
+        # check if arrays can be concatenated
+        _check_compatibile_arrays(ma, existing_array, append_axis)
+
+        # determine number of existing chunks along the append axis
+        existing_num_chunks = num_chunks(
+            array=group[name],
+            axis=append_axis,
+        )
 
-    # creates array if it doesn't already exist
-    arr = group.require_array(
-        name=name,
-        shape=zarray.shape,
-        chunk_shape=zarray.chunks,
-        dtype=encode_dtype(zarray.dtype),
-        codecs=zarray._v3_codec_pipeline(),
-        dimension_names=var.dims,
-        fill_value=zarray.fill_value,
-        # TODO fill_value?
-    )
-
-    # TODO it would be nice if we could assign directly to the .attrs property
-    for k, v in var.attrs.items():
-        arr.attrs[k] = encode_zarr_attr_value(v)
-    arr.attrs["_ARRAY_DIMENSIONS"] = encode_zarr_attr_value(var.dims)
+        # resize the array
+        arr = resize_array(
+            group=group,
+            name=name,
+            var=var,
+            append_axis=append_axis,
+        )
+    else:
+        # create array if it doesn't already exist
+        arr = group.require_array(
+            name=name,
+            shape=zarray.shape,
+            chunk_shape=zarray.chunks,
+            dtype=encode_dtype(zarray.dtype),
+            codecs=zarray._v3_codec_pipeline(),
+            dimension_names=var.dims,
+            fill_value=zarray.fill_value,
+            # TODO fill_value?
+        )
 
-    _encoding_keys = {"_FillValue", "missing_value", "scale_factor", "add_offset"}
-    for k, v in var.encoding.items():
-        if k in _encoding_keys:
+        # TODO it would be nice if we could assign directly to the .attrs property
+        # Aimee: Can we assume that the attributes are the same for the new array?
+        for k, v in var.attrs.items():
             arr.attrs[k] = encode_zarr_attr_value(v)
+        arr.attrs["_ARRAY_DIMENSIONS"] = encode_zarr_attr_value(var.dims)
+
+        _encoding_keys = {"_FillValue", "missing_value", "scale_factor", "add_offset"}
+        for k, v in var.encoding.items():
+            if k in _encoding_keys:
+                arr.attrs[k] = encode_zarr_attr_value(v)
 
     write_manifest_virtual_refs(
         store=store,
         group=group,
         arr_name=name,
         manifest=ma.manifest,
+        append_axis=append_axis,
+        existing_num_chunks=existing_num_chunks,
     )
 
 
+def generate_chunk_key(
+    index: np.nditer.multi_index,
+    append_axis: Optional[int] = None,
+    existing_num_chunks: Optional[int] = None,
+) -> str:
+    if append_axis is not None:
+        list_index = list(index)
+        # Offset by the number of existing chunks on the append axis
+        list_index[append_axis] += existing_num_chunks
+        index = tuple(list_index)
+    return "/".join(str(i) for i in index)
```

Comment:
I believe that creating mutable lists then altering them in-place is a bit more expensive than just creating the immutable tuple you want in one go. I think instead you could just do something like

```python
return "/".join(
    str(ind + existing_num_chunks) if axis is append_axis else str(ind) for axis, ind in enumerate(index)
)
```